code/scripting/api/libs/parse.cpp

```diff
@@ -0,0 +1,205 @@
+//
+//
+
+#include "parse.h"
+#include "cfile.h"
+#include "cfile/cfile.h"
+#include "parse/parselo.h"
+
+namespace scripting {
+namespace api {
+
+ADE_LIB(l_Parsing, "Parsing", "parse", "Engine parsing library");
+
+ADE_FUNC(readFileText, l_Parsing, "string file, string directory = <any>",
+         "Reads the text of the given file into the parsing system. If a directory is given then the file is read from "
+         "that location.",
+         "boolean", "true if the operation was successful, false otherwise")
+{
+	const char* file;
```

Comment:
It may be necessary to call pause_parse/unpause_parse here if the parser already has a file loaded.

Reply:
I am pretty sure that there is no case where a scripting hook is called while another file is being parsed since the first hook (Game Init) is called after all data has been parsed.

Also, `pause_parse` would require us to manage the text data ourself instead of letting the parse system handle that.